src/rust_connection/fd_read_write.rs

```diff
@@ -41,23 +57,27 @@ pub trait WriteFD {
     /// time.
     fn write_all(&mut self, mut buf: &[u8], mut fds: Vec<RawFdContainer>) -> Result<()> {
         while !buf.is_empty() || !fds.is_empty() {
-            let old_fds_len = fds.len();
-            match self.write(buf, &mut fds) {
-                Ok(0) if fds.len() == old_fds_len => {
-                    return Err(Error::new(
-                        ErrorKind::WriteZero,
-                        "failed to write whole buffer",
-                    ));
+            let (_, can_write) = self.poll(false, true, true)?;
+            if can_write {
```

Comment:
This just ignores readability? But then we are back to "we must also read while writing, but we don't do that", aren't we?

So far I thought that something like `write_all` is not possible on `WriteFD` and we'd need to handle this "write again" on some higher level.

Reply:
> This just ignores readability? But then we are back to "we must also read while writing, but we don't do that", aren't we?

`write_all` is only used in `write_setup`, where we don't need to follow that rule